source/exercises/mailroom.rst

```diff
@@ -17,6 +17,145 @@ incredibly boring, repetitive emails thanking your donors for their generous
 gifts. You are tired of doing this over and over again, so you've decided to
 let Python help you out of a jam and do your work for you.
 
+Intro Tutorial
+--------------
+
+Controlling Main Program Flow
+..............................
+
+One of the key components of the mailroom program is managing program flow and interacting with the user. Ideally main flow code should be cleanly separate from your feature code.
+
+The best way to manage the program flow is to use a ``while True`` loop which means you will keep asking the user for input until user selects a feature or exits.
+
+There are several ways to write your main program flow. Let's consider these two options:
+
+
+.. code-block:: python
+
+    def do_something():
+        # do things
+
+    def main():
+        while True:
+            do_something()
+
+    main()
+
+
+
+
+.. code-block:: python
+
+    def do_something()
+        # do things
+        main()
+
+    def main():
+        do_something()
+
+    main()
+
+
+Can you see the advantages of one over the other?
+In the first one, ``do_something`` is not aware of how the main works and as you add more features they shouldn't manage the main either.
+Another advantage is simpler code logic, and simpler code logic means less bugs!
```

Comment:
the other issue is that as do_something and main keep calling each-other, the call stack keeps getting deeper and deeper.